app/src/features/FhirResourceTree/resourceTreeSlice.ts

```diff
@@ -52,11 +63,67 @@ const resourceTreeSlice = createSlice({
         if (node) node.children = data.children;
       }
     },
+    setAttributeNodes: (
+      state,
+      { payload }: PayloadAction<{ attributeNodes: ElementNode[] }>
+    ) => {
+      const { attributeNodes } = payload;
+      const { attributeNodesInTree, root } = state;
+      if (root) {
+        const nodesToRemove = differenceBy(
+          attributeNodesInTree,
+          attributeNodes,
+          (node) => node.path
+        );
+        const nodesToAdd = differenceBy(
+          attributeNodes,
+          attributeNodesInTree,
+          (node) => node.path
+        );
+
+        // Add attribute nodes to the tree
+        nodesToAdd.forEach((node) => {
+          const parent = getNode(
+            "path",
+            computePathWithoutIndexes(node),
+            root as ElementNode
+          );
+          if (
+            parent &&
+            !parent.children.some(({ path }) => path === node.path)
+          ) {
+            parent.children.push(node);
+            attributeNodesInTree.push(node);
+          }
+        });
+
+        // Remove attribute nodes from the tree
+        nodesToRemove.forEach((node) => {
+          const parent = getNode(
+            "path",
+            computePathWithoutIndexes(node),
+            root as ElementNode
```

Comment:
would you need to do that using the "builder callback" notation https://redux-toolkit.js.org/usage/usage-with-typescript#type-safety-with-extrareducers ?

Reply:
I don't really get what you mean? This is already strongly typed right? 
What would it bring in this case? Because those actions are only being listened by this specific slice

Reply:
i was just wondering why you needed to cast `root` here!

Reply:
Oh ! Well even though root is tested not to be undefined, there were a warning saying it could be

Reply:
Well nvm.. Seems like the warning's gone, so i'll get rid of the cast